python/lsst/afw/image/readers.cc

```diff
@@ -202,153 +203,163 @@ void declareMultiPlaneMethods(py::class_<Class, Args...> & cls) {
-            );
-        },
-        "bbox"_a=lsst::geom::Box2I(), "origin"_a=PARENT, "conformMasks"_a=false, "allowUnsafe"_a=false,
-        "dtype"_a=py::none()
-    );
-    cls.def(
-        "read",
-        [](ExposureFitsReader & self, lsst::geom::Box2I const & bbox, ImageOrigin origin,
-           bool conformMasks, bool allowUnsafe, py::object dtype) {
-            if (dtype.is(py::none())) {
-                dtype = py::dtype(self.readImageDType());
-            }
-            return utils::python::TemplateInvoker().apply(
-                [&](auto t) {
-                    return self.read<decltype(t)>(bbox, origin, conformMasks, allowUnsafe);
+                "bbox"_a = lsst::geom::Box2I(), "origin"_a = PARENT, "conformMasks"_a = false, "allowUnsafe"_a = false,
+                "dtype"_a = py::none()
+        );
+        cls.def(
+                "read",
+                [](ExposureFitsReader &self, lsst::geom::Box2I const &bbox, ImageOrigin origin,
+                   bool conformMasks, bool allowUnsafe, py::object dtype) {
+                    if (dtype.is(py::none())) {
+                        dtype = py::dtype(self.readImageDType());
+                    }
+                    return utils::python::TemplateInvoker().apply(
+                            [&](auto t) {
+                                return self.read<decltype(t)>(bbox, origin, conformMasks, allowUnsafe);
+                            },
+                            py::dtype(dtype),
+                            utils::python::TemplateInvoker::Tag<std::uint16_t, int, float, double, std::uint64_t>()
+                    );
                 },
-                py::dtype(dtype),
-                utils::python::TemplateInvoker::Tag<std::uint16_t, int, float, double, std::uint64_t>()
-            );
-        },
-        "bbox"_a=lsst::geom::Box2I(), "origin"_a=PARENT, "conformMasks"_a=false, "allowUnsafe"_a=false,
-        "dtype"_a=py::none()
-    );
+                "bbox"_a = lsst::geom::Box2I(), "origin"_a = PARENT, "conformMasks"_a = false, "allowUnsafe"_a = false,
+                "dtype"_a = py::none()
+        );
+    });
 }
 
 
 PYBIND11_MODULE(readers, mod) {
-    py::module::import("lsst.daf.base");
-    py::module::import("lsst.geom");
-    py::module::import("lsst.afw.image.image");
-    py::module::import("lsst.afw.image.maskedImage");
-    py::module::import("lsst.afw.image.exposure");
-    declareImageFitsReader(mod);
-    declareMaskFitsReader(mod);
-    declareMaskedImageFitsReader(mod);
-    declareExposureFitsReader(mod);
+    lsst::utils::python::WrapperCollection wrappers(mod, "lsst.afw.image.readers");
+    wrappers.addInheritanceDependency("lsst.daf.base");
+    wrappers.addInheritanceDependency("lsst.geom");
+    wrappers.addInheritanceDependency("lsst.afw.image.image");
+    wrappers.addInheritanceDependency("lsst.afw.image.maskedImage");
+    wrappers.addInheritanceDependency("lsst.afw.image.exposure");
```

Comment:
Signature dependencies should work for all of these.

Reply:
Same as above